File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        
        res = [0] * len(nums)
        i = len(res)-1
        left = 0
        right = len(nums)-1
        
        while left<=right:

            if abs(nums[left]) > abs(nums[right]):
                res[i] = nums[left]**2
                left+=1
                i-=1
            else:
                res[i] = nums[right]**2
                i-=1
                right-=1
        return res
            

File: 1019-squares-of-a-sorted-array/1019-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> array(nums.size());
        int left = 0;
        int right = nums.size() - 1;
        int idx = nums.size()-1;

        while (left<=right){
            if (abs(nums[left]) > abs(nums[right])){
                array[idx] = nums[left]*nums[left];
                left++;
            }
            else{
                array[idx] = nums[right]*nums[right];
                right--;
            }
            idx--;

        }

        return array;
    }
};